### ISSUE TITLE
Pickadate dialog dosen't close on select bug

### ISSUE THREAD

**Issue by anik786** [amsul/pickadate.js#924]
I have the following basic code:

```
<!doctype html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport"
          content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
    <title>Document</title>
    <link rel="stylesheet" href="/bower_components/pickadate/lib/compressed/themes/default.css">
    <link rel="stylesheet" href="/bower_components/pickadate/lib/compressed/themes/default.date.css">
    <link rel="stylesheet" href="/bower_components/pickadate/lib/compressed/themes/default.time.css">
</head>
<body>

<label>
    Start Date:
    <input type="date" name="start_date" class="datepicker">
</label>

<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
<script src="/bower_components/pickadate/lib/picker.js"></script>
<script src="/bower_components/pickadate/lib/picker.date.js"></script>
<script>
    $('.datepicker').pickadate({
        closeOnSelect: true, // should be default
    });
</script>
</body>
</html>
```

This code seems to work fine and has no errors on the console.
The only issue is when I pick a date, the dialog dosent close automatically like it does on the website: http://amsul.ca/pickadate.js/date/

I am using:
 
Datepicker: v3.5.6 
OS: Windows 10
Browser: 53.0.2785.143 (Official Build) m (64-bit)

**Comment by anik786** [comment]
OMG! I have spent 1 hour and 15min on this I finally figured out when this bug happens!

It occurs when the input is wrapped around a label with no for tag.

Like so:

```
    <label>
        <input class="datepicker" type="date">
    </label>
```

However this works fine

````
    <label for="">
        <input class="datepicker" type="date">
    </label>
````
 Please fix this bug!

**Comment by bl4ckdu5t** [comment]
From the look of it it's not a bug. It simply just expects you to write valid HTML. I personally never have `<label>` without a `for` attribute and an id for its reference on the input element.

**Comment by anik786** [comment]
Actually this is perfectly valid html! According to W3 standards, if you have a label with an input inside of it, the input is associated with that label as shown here https://www.w3.org/wiki/HTML/Elements/label#Example_B

Test it yourself: https://validator.w3.org/#validate_by_input

It's also a common practice!

**Comment by anik786** [comment]
I have now pinpointed exactly when this bug occurs.

Basically, the bug occurs when an input is wrapped in a label with no "for" attribute like so:

````
<label>
       DOB:
       <input class="datepicker" type="date">
</label>
````
OR

if the id of the label matches the input, while the input is a child element of the label like so:

````
<label for="foo">
       DOB:
        <input class="datepicker" type="date" id="foo">
</label>
````

Both of the above HTML snippets is W3 compliant.

Note the bug does NOT occur when the for tag does not match the id (NOT W3 compliant):

````
<label for="bar">
       DOB:
        <input class="datepicker" type="date" id="foo">
</label>
````

or if the input is not a child element of the label (Is W3 compliant):

````
<label for="foo">DOB:</label>
<input class="datepicker" type="date" id="foo">
````

**Comment by bl4ckdu5t** [comment]
That wouldn't be an issue with the next major version release

**Comment by anik786** [comment]
Great. But why isn't this bug fix being added to this release. I don't think it's a breaking feature. Surely it can be added as a minor version, as it's just a bug fix.

**Comment by bl4ckdu5t** [comment]
This version has no more minor versions for now but yes it may be considered as a thing for a patch version

**Comment by amsul** [comment]
Yup, @bl4ckdu5t is right. Plus, there is a solution mentioned in the docs: http://amsul.ca/pickadate.js/date/#container
